spec/support/synchronization_helpers.rb

```diff
@@ -53,11 +53,23 @@ def expect_in_thread(&block)
 
   # Defaults to 5 second timeout
   def try_wait_until(attempts: 50, backoff: 0.1)
```

Comment:
This is a significant improvement, as before `try_wait_until` would always immediately check for the provided condition (through `yield`) but that's normally not enough time for whatever background job to run.

So that `yield` is normally wasted. But at least it's cheap, so we do it because when it's fulfilled right way it saves quite a bit of time.

But the main issue here is that if that early check fails, we go straight into a `sleep(100ms)`, and that's super common.

So we replace the first sleep with a `Thread.pass`, which normally allows for background threads to start and, if they are short enough, execute the desired work before we perform another `yield` check.

This saves quite a bit of time across the code base because this 1 time 100ms sleep is the de facto most common case for this method.